models/quality_measures/intermediate/cqm438_statin_therapy_for_patients_with_cardiovascular_disease/quality_measures__int_cqm438_denominator.sql

```diff
@@ -0,0 +1,264 @@
+    from visits_encounters
+
+    union all
+
+    select *, cast(null as {{ dbt.type_string() }}) as visit_enc, 'p' as proc_enc, cast(null as {{ dbt.type_string() }}) as claim_enc
+    from procedure_encounters
+
+    union all
+    
+    select *, cast(null as {{ dbt.type_string() }}) as visit_enc,cast(null as {{ dbt.type_string() }}) as proc_enc, 'c' as claim_enc
+    from claims_encounters
+
+)
+
+, encounters_by_patient as (
+
+    select patient_id, min(min_date) min_date, max(max_date) max_date,
+        concat(concat(
+            coalesce(min(visit_enc),'')
+            ,coalesce(min(proc_enc),''))
+            ,coalesce(min(claim_enc),'')
+            ) as qualifying_types
+    from all_encounters
+    group by patient_id
+
+)
+
+, patients_with_age as (
+
+    select
+          p.patient_id
+        , floor({{ datediff('birth_date', 'performance_period_begin', 'hour') }} / 8760.0)  as age
+    from {{ref('quality_measures__stg_core__patient')}} p
+    inner join encounters_by_patient e
+        on p.patient_id = e.patient_id
+            and p.death_date is null
+    cross join {{ref('quality_measures__int_cqm438__performance_period')}}
+
+)
+
+, ascvd_codes as (
+
+    select
+          code
+        , code_system
+    from {{ ref('quality_measures__value_sets') }}
+    where lower(concept_name) in (
+              'atherosclerosis and peripheral arterial disease'
+            , 'myocardial infarction'
+            , 'pci'
+            , 'stable and unstable angina'
+            , 'atherosclerosis and peripheral arterial disease'
+            , 'cabg or pci procedure'
+            , 'cabg surgeries'
+            , 'cerebrovascular disease stroke or tia'
+            , 'ischemic heart disease or related diagnoses'
+        )
+
+)
+
+, conditions as (
+
+    select
+          patient_id
+        , claim_id
+        , encounter_id
+        , recorded_date
+        , source_code
+        , source_code_type
+        , normalized_code
+        , normalized_code_type
+    from {{ ref('quality_measures__stg_core__condition')}}
+
+)
+
+, ascvd_conditions as (
+
+    select
+          conditions.patient_id
+        , conditions.recorded_date as evidence_date
+    from conditions
+    inner join ascvd_codes
+        on conditions.source_code_type = ascvd_codes.code_system
+            and conditions.source_code = ascvd_codes.code
+
+)
+
+, procedures as (
+
+    select
+          patient_id
+        , procedure_date
+        , coalesce (
+              normalized_code_type
+            , case
+                when lower(source_code_type) = 'cpt' then 'hcpcs'
+                when lower(source_code_type) = 'snomed' then 'snomed-ct'
+                else lower(source_code_type)
+              end
+          ) as code_type
+        , coalesce (
+              normalized_code
+            , source_code
+          ) as code
+    from {{ ref('quality_measures__stg_core__procedure') }}
```

Comment:
perform such transformations in earlier `procedure_encounters` cte and reference here